lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -583,7 +587,13 @@ class MongoClientInterface {
         }, {
             upsert: true,
         })
-            .then(() => cb(null))
+            .then(result => {
+                if (result.modifiedCount === 0 && result.upsertedCount === 0) {
```

Comment:
is it really an error?
the update was useless, but we "reached" the expected state, and upper layer can proceed...

Reply:
There should be an additional `result.matchedCount` here, meaning that we really missed the update...
But actually, why do we accept upserts in this case? We should already have a bucketMD when we update its attributes...